holoviews/plotting/bokeh/callbacks.py

```diff
@@ -249,8 +303,12 @@ class RangeXYCallback(Callback):
     handles = ['x_range', 'y_range']
 
     def _process_msg(self, msg):
-        return {'x_range': (msg['x0'], msg['x1']),
-                'y_range': (msg['y0'], msg['y1'])}
+        data = {}
+        if 'x0' in msg and 'x1' in msg:
+            data['x_range'] = (msg['x0'], msg['x1'])
+        if 'y0' in msg and 'y1' in msg:
+            data['y_range'] = (msg['y0'], msg['y1'])
+        return data
 
 
 class RangeXCallback(Callback):
```

Comment:
All of the new bits of code below seems to have the form:

```
if predicate(msg):
   return dictionary_from_msg(msg)
else:
  return {}
```

If you agree this is a general pattern, maybe we can just have an ``applicable`` predicate method with the baseclass checking the predicate value. E.g

```python
class RangeXCallback(Callback):
      handles = ['x_range']

     def applicable(msg):
          return 'x0' in msg and 'x1' in msg
  
      def _process_msg(self, msg):
         return {'x_range': (msg['x0'], msg['x1'])}
```

And of course ``applicable`` can return ``True`` by default.